src/Controller/CRUDController.php

```diff
@@ -66,9 +118,28 @@ class CRUDController extends Controller
      */
     private $templateRegistry;
 
-    public function setContainer(?ContainerInterface $container = null): void
-    {
-        $this->container = $container;
+    public function __construct(
```

Comment:
If the developper needs another dependency, he won't be able to use the construct without adding all these dependency.

And if he just pass the mandatory arguments, he'll get the error for exportAction "You need to install exporter" which can be already installed.
And if he wants to pass the optional arguments, I'm not sure it's easy to know what is expected as `$userManager` since the type is `object`. 

Maybe it could be a nice idea to use this instead: https://symfony.com/doc/current/service_container/autowiring.html#autowiring-other-methods-e-g-setters-and-public-typed-properties 

WDYT ?

Reply:
I don't know if there is a good solution TBH. Having setters and getters means I have to wrap all the calls with null !== $property, because that setter could have not been called.

Another option is to recieve all dependencies directly on the methods, but not sure is a good solution either, we could end up will long method calls because of that. (and a mess on method parameters)

Reply:
> Another option is to recieve all dependencies directly on the methods, but not sure is a good solution either, we could end up will long method calls because of that. (and a mess on method parameters)

This does not sounds like a good idea to me since the developper could override some action. It would be useless to use the same method definition (with all the previously needed dependencies) if all these dependencies are not used anymore.

@sonata-project/contributors Any idea on this ?

Reply:
I don't know it would work, but since we are extending `AbstractController`, what about to use the same strategy they use, so we leave the calls to `has` and `get` and just override the `getSubscribedServices`.

Ref: https://symfony.com/doc/4.4/service_container/service_subscribers_locators.html#including-services

Reply:
To use the getSubscribedServices, the only way I got it working is to use autowire, otherwise my injected container has everything (public). I like the idea but not sure if we like using autowire.

That was another idea I was about to write: we can just inject container and leave the old logic as is, with the benefit to move forward with sf5.

Reply:
Maybe [adding `container.service_subscriber` tag](https://symfony.com/doc/4.4/service_container/service_subscribers_locators.html#defining-a-service-subscriber).

Reply:
Already added, it does not restrict what I can get from container. And if I remove the setContainer I get a message about missing that, which is not missing

Reply:
> Already added, it does not restrict what I can get from container. And if I remove the setContainer I get a message about missing tag, which is not missing

If we keep the container property, it could be `private` now